Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM python:3.9.18-alpine3.18 as base
+
+RUN pip install "poetry==1.3.2"
+
+COPY ./todo_app ./todo_app
+COPY ./poetry.lock .
+COPY pyproject.toml .
+
+RUN poetry install
```

Comment:
Minor, but if you're regularly rebuilding the app locally you might find a small optimisation in reordering these lines to:
```dockerfile
COPY ./poetry.lock .
COPY pyproject.toml .

RUN poetry install
COPY ./todo_app ./todo_app
```

Since this way we only need to do the reasonably-slow install step when we change our dependencies, not when we just change application code